opc/redis/streams.go

```diff
@@ -0,0 +1,265 @@
+package redis
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/go-redis/redis"
+	"github.com/mainflux/mainflux/logger"
+	"github.com/mainflux/mainflux/opc"
+)
+
+const (
+	protocol = "opc"
+
+	group  = "mainflux.opc"
+	stream = "mainflux.things"
+
+	thingPrefix = "thing."
+	thingCreate = thingPrefix + "create"
+	thingUpdate = thingPrefix + "update"
+	thingRemove = thingPrefix + "remove"
+
+	channelPrefix = "channel."
+	channelCreate = channelPrefix + "create"
+	channelUpdate = channelPrefix + "update"
+	channelRemove = channelPrefix + "remove"
+
+	exists = "BUSYGROUP Consumer Group name already exists"
+)
+
+var (
+	errMetadataType = errors.New("metadatada is not of type opc")
+
+	errMetadataAppID = errors.New("application ID not found in channel metadatada")
+
+	errMetadataDevEUI = errors.New("device EUI not found in thing metadatada")
+)
+
+// EventStore represents event source for things and channels provisioning.
+type EventStore interface {
```

Comment:
You can rename this one to the `Subscriber` too. But here it even makes sense to use this name (I feel like here interface API could change over time). This about moving all of these interfaces (without implementations) to the `opc` package. If you do this, I know that you'll have a name collision. In order to avoid it, you can rename your interfaces from `Subscriber` to some other names that are not so tech-specific (try avoiding putting protocol and lib names into the interface name).